### PR TITLE
EZEE-3337 - Image Editor configuration

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | EZEE-3337 |
| Versions      | master |

Add an article on basic IE configuration.

### CODE REVIEW

File: docs/guide/image_editor.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+The example sets the aspect ratio values and label names for buttons used by the Crop feature.
+
+
+``` yaml
+system:
+    <siteaccess_name>:
+        image_editor:
+            action_groups:
+                default:
+                    id: default
```
Comment (dew326) [inline docs/guide/image_editor.md:23]: I would mention that developers can create more than one group.

File: docs/guide/image_editor.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    label: Default
+                    actions:
+                        (...)
+                        crop:
```
Comment (dew326) [inline docs/guide/image_editor.md:27]: I would mention that each action can have `priority` and `visible` - thanks to that you can change the order of actions and remove some actions from the image editor.

File: docs/guide/image_editor.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# Configuring the Image Editor
+
+## Introduction
+
```
Comment (DominikaK) [inline docs/guide/image_editor.md:4]: Not neded
```suggestion
```

File: docs/guide/image_editor.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+When a Content Item contains Fields of the [ezimage](../api/field_type_reference.md#image-field-type) or [ezimageasset](../api/field_type_reference.md#imageasset-field-type) type, users can perform basic image editing functions with the Image Editor.
```
Comment (DominikaK) [inline docs/guide/image_editor.md:5]:
```suggestion
When a Content item contains Fields of the [ezimage](../api/field_type_reference.md#image-field-type) or [ezimageasset](../api/field_type_reference.md#imageasset-field-type) type, users can perform basic image editing functions with the Image Editor.
```

File: docs/guide/image_editor.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+For more information, see the [user documentation](https://doc.ibexa.co/projects/userguide/en/master/editing_images/).
+
+## Configuration
+
+You can modify the default settings to change the appearance or behaviour of the Image Editor.
```
Comment (DominikaK) [inline docs/guide/image_editor.md:10]: AmE
```suggestion
You can modify the default settings to change the appearance or behavior of the Image Editor.
```

File: docs/guide/image_editor.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+You can also expand the default set of parameters to create buttons that may be required by custom features that you add by [extending the Image Editor](#extending-the-image-editor).
+To do this, modify the `config/packages/ezplatform.yaml` file, or create a separate YAML file in the `config/packages` folder, and add a settings tree similar to the following example.
+Image Editor settings are [dynamic](config_dynamic.md), therefore you can configure a different set of buttons or preset values for each of the SiteAccesses in your installation.
```
Comment (DominikaK) [inline docs/guide/image_editor.md:13]: We use "SiteAccess-aware" throughout the rest of the doc, should be enough here:
```suggestion
Image Editor settings are SiteAccess-aware.
```

File: docs/guide/image_editor.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+The example sets the aspect ratio values and label names for buttons used by the Crop feature.
+
```
Comment (DominikaK) [inline docs/guide/image_editor.md:15]:
```suggestion
```

File: docs/guide/image_editor.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+``` yaml
+system:
+    <siteaccess_name>:
```
Comment (DominikaK) [inline docs/guide/image_editor.md:19]:
```suggestion
    <siteaccess>:
```

File: docs/guide/image_editor.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                                        x: 16
+                                        y: 9
+                                custom:
+                                    label: Custom
+```
+
+### Additional information
+
+Each image can be accompanied by additional information that is not visible to the user.
+By default, additional information serves to store the coordinates of the [focal point](https://doc.ibexa.co/projects/userguide/en/master/editing_images/#focal-point), but you can use this extension point to pass various parameters of custom features that you add by [extending the Image Editor](#extending-the-image-editor).
```
Comment (DominikaK) [inline docs/guide/image_editor.md:57]:
```suggestion
By default, additional information stores the coordinates of the [focal point](https://doc.ibexa.co/projects/userguide/en/master/editing_images/#focal-point),
but you can use this extension point to pass various parameters of custom features
that you add by [extending the Image Editor](#extending-the-image-editor).
```

File: docs/guide/image_editor.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+             'author' => 'John Smith',
+         ],
+     ],
+ ]),
+```
+ 
+## Extending the Image Editor
+
+You can extend the Image Editor if your users require additional features.
+For example, you can create a feature that modifies the colour palette of an image.
```
Comment (DominikaK) [inline docs/guide/image_editor.md:82]:
```suggestion
For example, you can create a feature that modifies the color palette of an image.
```

File: docs/guide/image_editor.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+         'additionalData' => [
+             'focalPointX' => 50,
+             'focalPointY' => 100,
+             'author' => 'John Smith',
+         ],
+     ],
+ ]),
+```
+ 
+## Extending the Image Editor
```
Comment (DominikaK) [inline docs/guide/image_editor.md:79]: Not sure if it makes sense to include this paragraph, if we're not describing the procedure in detail.
If this is is something we should have a detailed guide on, let's add is as a follow-up.
Then also remove the link above.

File: docs/guide/image_editor.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+## Introduction
+
+When a Content Item contains Fields of the [ezimage](../api/field_type_reference.md#image-field-type) or [ezimageasset](../api/field_type_reference.md#imageasset-field-type) type, users can perform basic image editing functions with the Image Editor.
+For more information, see the [user documentation](https://doc.ibexa.co/projects/userguide/en/master/editing_images/).
+
+## Configuration
+
+You can modify the default settings to change the appearance or behaviour of the Image Editor.
+You can also expand the default set of parameters to create buttons that may be required by custom features that you add by [extending the Image Editor](#extending-the-image-editor).
+To do this, modify the `config/packages/ezplatform.yaml` file, or create a separate YAML file in the `config/packages` folder, and add a settings tree similar to the following example.
```
Comment (DominikaK) [inline docs/guide/image_editor.md:12]:
```suggestion

To do this, modify the `config/packages/ezplatform.yaml` file, or create a separate YAML file in the `config/packages` folder, and add a settings tree similar to the following example.
```

File: docs/guide/image_editor.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+Image Editor settings are [dynamic](config_dynamic.md), therefore you can configure a different set of buttons or preset values for each of the SiteAccesses in your installation.
+The example sets the aspect ratio values and label names for buttons used by the Crop feature.
```
Comment (DominikaK) [inline docs/guide/image_editor.md:14]:
```suggestion

The following example sets the aspect ratio values and label names for buttons used by the Crop feature.
```

**Review by DominikaK — CHANGES_REQUESTED** [review]

File: docs/guide/image_editor.md (hunk 4 above)
Comment (ViniTou) [reply docs/guide/image_editor.md:5]: I think it is worth mentioning that its not working on DAM assets atm (button would be disabled/invisible).

File: docs/guide/image_editor.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
+# Configuring the Image Editor
```
Comment (DominikaK) [inline docs/guide/image_editor.md:1]: Please add an Experience/Commerce badge, as https://github.com/ezsystems/developer-documentation/pull/1230 has been merged and image editor is in those two editions only.

**Review by dew326 — APPROVED** [review]

**Review by DominikaK — APPROVED** [review]

**Pull request merged**